src/Compilers/CSharp/Portable/BoundTree/BoundDagEvaluation.cs

```diff
@@ -1,15 +1,22 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     partial class BoundDagEvaluation
     {
+        private static IEqualityComparer<ISymbol> s_ignoreEverythingComparer = SymbolEqualityComparer.IgnoreEverything;
+
         public override bool Equals(object obj) => obj is BoundDagEvaluation other && this.Equals(other);
         public virtual bool Equals(BoundDagEvaluation other)
         {
-            return other != (object)null && this.Kind == other.Kind && this.GetOriginalInput().Equals(other.GetOriginalInput()) && this.Symbol == other.Symbol;
+            return other != (object)null &&
+                this.Kind == other.Kind &&
+                this.GetOriginalInput().Equals(other.GetOriginalInput()) &&
+                s_ignoreEverythingComparer.Equals(this.Symbol?.GetPublicSymbol(), other.Symbol?.GetPublicSymbol());
```

Comment:
>s_ignoreEverythingComparer.Equals(this.Symbol?.GetPublicSymbol(), other.Symbol?.GetPublicSymbol() [](start = 16, length = 97)

Do we need to use `SymbolEqualityComparer` and `GetPublicSymbol()` or could this be `Symbol.Equals(this.Symbol, other.Symbol, TypeCompareKind.AllIgnoreOptions)`?